app/code/core/Mage/Admin/Model/Observer.php

```diff
@@ -100,11 +100,19 @@ public function actionPreDispatchAdmin($observer)
                             ->setActionName('deniedJson')
                             ->setDispatched(false);
                     } else {
-                        $request->setParam('forwarded', true)
-                            ->setRouteName('adminhtml')
-                            ->setControllerName('index')
-                            ->setActionName('login')
-                            ->setDispatched(false);
+                        if ($request->getRouteName() == 'adminhtml') {
```

Comment:
What about alternatively using `Mage::app()->getStore()->isAdmin()` ? This way older routes that are not in the adminhtml namespace would still work.